doc/threat_model/threat_model.md

```diff
@@ -216,6 +231,37 @@ The FPGA design as well as the firmware has been audited, and
 hardening of these has been performed to some degree. For more
 information, see the [Release Notes](/doc/release_notes.md)
 
+#### Mitigations
+
+- To protect the UDS the hardware design allows only one read per word
+  of the UDS per power-cycle.
+
+- USB port attacks - boot protocol:
+
+  - The firmware has a more strict protocol state machine and exits out
+    to an eternal loop on any errors.
+
+  - Firmware stack is protected by hardware for execution.
+
+- Software attacks:
+
+  - Firmware uses its own FW_RAM for sensitive computations which is
+    not available in app mode.
+
+  - Device apps can protect arbitrarly parts of RAM, typically heap +
+    stack, with hardware support.
+
+- Hardware attacks:
+
+  - The reading and handling of the UDS is randomized so it doesn't
+    always occur on the same cycle.
+
+  - Device apps can turn on hardware assisted RAM address and data
```

Comment:
Isn't it firmware? Or am I missing something.. RAM_ASLR and RAM_SCRAMBLE in MMIO table are invisible in app-mode.

Reply:
You're right. I'll change.